lib/chef/provider/package.rb

```diff
@@ -337,7 +365,7 @@ def target_version_already_installed?(current_version, target_version)
       # `version_satisfied_by?(version, constraint)` might be a better name to make this generic.
       #
       def version_requirement_satisfied?(current_version, new_version)
-        target_version_already_installed?(current_version, new_version)
+        version_equals?(current_version, new_version)
```

Comment:
i think you still have to call the old version here so that overriding it works.  although you might want to `responds_to?` it here, and throw a deprecation warning and call it, then call `version_equals?` -- then actually delete or comment out the definition of the method here entirely.